spark/src/main/scala/org/apache/mahout/drivers/RowSimilarityDriver.scala

```diff
@@ -135,7 +133,7 @@ object RowSimilarityDriver extends MahoutSparkDriver {
   }
 
   override def process: Unit = {
-    start()
+    start
```

Comment:
quoting from your reference:
`However, this syntax should *only* be used when the method in question has
no side-effects (purely-functional). In other words, it would be acceptable
to omit parentheses when calling queue.size, but not when calling println().
`

Kind of what i am saying. No side effects rule is violated (and, frankly,
it is violated by anything but a computed property).

On Wed, Feb 18, 2015 at 4:33 PM, Pat Ferrel <notifications@github.com>
wrote:

> In
> spark/src/main/scala/org/apache/mahout/drivers/RowSimilarityDriver.scala
> <https://github.com/apache/mahout/pull/76#discussion_r24956351>:
>
> > @@ -135,7 +133,7 @@ object RowSimilarityDriver extends MahoutSparkDriver {
> >    }
> >
> >    override def process: Unit = {
> > -    start()
> > +    start
>
> The function is an Arity-0 method. Perhaps I misread:
> http://docs.scala-lang.org/style/method-invocation.html
>
> Arity-0
> Scala allows the omission of parentheses on methods of arity-0 (no
> arguments):
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/mahout/pull/76/files#r24956351>.
>